internal/fswatcher/os_others.go

```diff
@@ -46,11 +46,8 @@ func (w *fswatcher) populateWatches(initPath string, filter watchFilterFn) ([]st
 			return nil
 		}
 
-		if info.IsDir() && filter(path) {
-			// We might end up here if the user creates a go.mod in a watched dir
-			// for example. Then we need to remove the watch.
+		if !selectPath(path, info) {
```

Comment:
The original `info.IsDir()` here admittedly looks a bit weird, but IIRC it can't just be dropped.
I can't remember on top of my head why. Might it be related to the `return filepath.SkipDir` that we (if I understand it correctly) don't want to return for a file.

Reply:
done